tests/server.rs

```diff
@@ -353,6 +353,66 @@ mod response_body_lengths {
             expects_con_len: false,
         });
     }
+
+    #[test]
+    #[ignore]
+    fn http2_auto_response_with_known_length() {
+        // TODO: check that this really works after #1546
+        use hyper::body::Payload;
+
+        let server = serve();
+        let addr_str = format!("http://{}", server.addr());
+        server.reply().body("Hello, World!");
+
+        hyper::rt::run(hyper::rt::lazy(move || {
+            let client: Client<_, hyper::Body> = Client::builder().http2_only(true).build_http();
+            let uri = addr_str
+                .parse::<hyper::Uri>()
+                .expect("server addr should parse");
+
+            client
+                .get(uri)
+                .and_then(|res| {
+                    assert_eq!(res.body().content_length(), Some(13));
```

Comment:
At least the header can be checked for, right? `assert_eq!(res.headers()["content-length"], "13")`.

With that, these `#[ignore]`s could be removed, and just the checking of `res.body().content_length()` could be commented out with a TODO.

Reply:
Done